x-pack/plugins/cases/public/containers/utils.ts

```diff
@@ -30,20 +30,80 @@ import {
   CaseUserActionsResponseRt,
   CommentType,
   CasePatchRequest,
+  CaseUserActionConnectorRt,
+  CaseConnector,
+  ConnectorTypes,
+  noneConnectorId,
+  CaseFullExternalService,
+  CaseUserActionExternalServiceRt,
 } from '../../common';
 import { AllCases, Case, UpdateByKey } from './types';
 import * as i18n from './translations';
 
 export const getTypedPayload = <T>(a: unknown): T => a as T;
 
-export const parseString = (params: string) => {
+export const parseStringAsConnector = (
+  id: string | null,
+  encodedData: string | null
+): CaseConnector | undefined => {
+  if (encodedData == null) {
+    return;
+  }
+
+  const decodedConnector = parseString(encodedData);
+
+  if (!CaseUserActionConnectorRt.is(decodedConnector)) {
+    return;
+  }
+
+  if (id == null && decodedConnector.type === ConnectorTypes.none) {
+    return {
+      ...decodedConnector,
+      id: noneConnectorId,
+    };
+  } else if (id == null) {
+    return;
+  } else {
+    // id does not equal null or undefined and the connector type does not equal none
+    // so return the connector with its id
+    return {
+      ...decodedConnector,
+      id,
+    };
+  }
+};
+
+const parseString = (params: string | null): unknown | null => {
```

Comment:
I removed this from being exported since it is only used within this file and moved it here.